src/node_contextify.cc

```diff
@@ -452,31 +366,107 @@ class ContextifyContext {
   }
 
 
-  static void GlobalPropertyQueryCallback(
+  static void GlobalPropertyDescriptorCallback(
       Local<Name> property,
-      const PropertyCallbackInfo<Integer>& args) {
+      const PropertyCallbackInfo<Value>& info) {
     ContextifyContext* ctx;
-    ASSIGN_OR_RETURN_UNWRAP(&ctx, args.Data().As<Object>());
+    ASSIGN_OR_RETURN_UNWRAP(&ctx, info.Data().As<Object>());
 
     // Still initializing
     if (ctx->context_.IsEmpty())
       return;
-
     Local<Context> context = ctx->context();
-    Maybe<PropertyAttribute> maybe_prop_attr =
-        ctx->sandbox()->GetRealNamedPropertyAttributes(context, property);
+    v8::Isolate* isolate = context->GetIsolate();
+    HandleScope scope(isolate);
 
-    if (maybe_prop_attr.IsNothing()) {
-      maybe_prop_attr =
-          ctx->global_proxy()->GetRealNamedPropertyAttributes(context,
-                                                              property);
-    }
+    Local<String> key = property->ToString(isolate);
+    Local<Object> sandbox = ctx->sandbox();
 
-    if (maybe_prop_attr.IsJust()) {
-      PropertyAttribute prop_attr = maybe_prop_attr.FromJust();
-      args.GetReturnValue().Set(prop_attr);
+    Maybe<bool> has = sandbox->HasOwnProperty(context, key);
+    Local<Value> descriptor_intercepted = has.IsNothing() ? ctx->global_proxy()
+        ->GetOwnPropertyDescriptor(context, key).ToLocalChecked() :
```

Comment:
I think this might be more readable if you had a line break after the `?`… if that makes the line too long, you can put the `.ToLocalCheck()` onto its own line (with 4 extra spaces), I guess